site/ts/twig/grading/electronic/SubmissionPanel.ts

```diff
@@ -0,0 +1,88 @@
+import { urlDecode } from '../../../utils/server';
+
+// expand all files in Submissions and Results section
+function openAll(click_class: string, class_modifier: string) {
+    const toClose = $(`#div_viewer_${$(`.${click_class}${class_modifier}`).attr('data-viewer_id')}`).hasClass('open');
+    $('#submission_browser').find(`.${click_class}${class_modifier}`).each(function () {
+        // Check that the file is not a PDF before clicking on it
+        const viewerID = $(this).attr('data-viewer_id');
+        if (($(this).parent().hasClass('file-viewer') && $(`#file_viewer_${viewerID}`).hasClass('shown') === toClose) ||
+            ($(this).parent().hasClass('div-viewer') && $(`#div_viewer_${viewerID}`).hasClass('open') === toClose)) {
+            const innerText = Object.values($(this))[0].innerText;
+            if (innerText.slice(-4) !== '.pdf') {
+                $(this).click();
+            }
```

Comment:
Is this the best way to check if it's a PDF and open the element?

Reply:
I think the best way to do this would be to additionally pass the MIME type over, but I think that might be better in a separate PR since this PR focuses mainly on updating the Twig/TS files.